src/cascadia/TerminalSettingsModel/ActionMap.cpp

```diff
@@ -193,39 +193,79 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
+                // ShortcutAction is GlobalSummon or QuakeMode
+                if (cmd.ActionAndArgs().Action() == ShortcutAction::GlobalSummon || cmd.ActionAndArgs().Action() == ShortcutAction::QuakeMode)
                 {
-                    // (2) haven't been visited already
-                    const auto actionID{ Hash(actionAndArgs) };
-                    if (visitedActionIDs.find(actionID) == visitedActionIDs.end())
-                    {
-                        const auto& cmdImpl{ get_self<Command>(cmd) };
-                        for (const auto& keys : cmdImpl->KeyMappings())
-                        {
-                            // (3) haven't had that key chord added yet
-                            if (globalHotkeys.find(keys) == globalHotkeys.end())
-                            {
-                                globalHotkeys.emplace(keys, cmd);
-                            }
-                        }
-
-                        // Record that we already handled adding this action to the NameMap.
-                        visitedActionIDs.emplace(actionID);
-                    }
+                    globalHotkeys.emplace(keys, cmd);
                 }
             }
             _GlobalHotkeysCache = single_threaded_map<Control::KeyChord, Model::Command>(std::move(globalHotkeys));
         }
         return _GlobalHotkeysCache.GetView();
     }
 
+    Windows::Foundation::Collections::IMapView<Control::KeyChord, Model::Command> ActionMap::KeyBindings()
+    {
+        if (!_KeyBindingMapCache)
+        {
+            // populate _KeyBindingMapCache
+            std::unordered_map<KeyChord, Model::Command, KeyChordHash, KeyChordEquality> keyBindingsMap;
+            std::unordered_set<KeyChord, KeyChordHash, KeyChordEquality> unboundKeys;
+            _PopulateKeyBindingMapWithStandardCommands(keyBindingsMap, unboundKeys);
+
+            _KeyBindingMapCache = single_threaded_map<KeyChord, Model::Command>(std::move(keyBindingsMap));
+        }
+        return _KeyBindingMapCache.GetView();
+    }
+
+    // Method Description:
+    // - Populates the provided keyBindingsMap with all of our actions and our parents actions
+    //    while omitting the key bindings that were already added before.
+    // - This needs to be a bottom up approach to ensure that we only add each key chord once.
+    // Arguments:
+    // - keyBindingsMap: the keyBindingsMap we're populating. This maps the key chord of a command to the command itself.
+    // - unboundKeys: a set of keys that are explicitly unbound
+    void ActionMap::_PopulateKeyBindingMapWithStandardCommands(std::unordered_map<KeyChord, Model::Command, KeyChordHash, KeyChordEquality>& keyBindingsMap, std::unordered_set<Control::KeyChord, KeyChordHash, KeyChordEquality>& unboundKeys) const
+    {
+        // Update KeyBindingsMap with our current layer
+        for (const auto& [keys, actionID] : _KeyMap)
+        {
+            const auto cmd{ _GetActionByID(actionID).value() };

```

Comment:
Whoa! This immediately unpacks the std::optional. I'm guessing that this is guaranteed?

Might be worth noting in the null check below that it captures the case where the optional was populated _with null_ (and what that means)